.azure-pipelines-templates/matrix.yml

```diff
@@ -92,6 +92,29 @@ jobs:
           ctest_filter: "${{ parameters.test.virtual_perf.ctest_args }}"
           depends_on: configure
 
+  - ${{ if eq(parameters.perf_tests, 'run') }}:
+      - template: common.yml
+        parameters:
+          target: Virtual
+          env: ${{ parameters.env.Virtual }}
+          cmake_args: "${{ parameters.build.common.cmake_args }} ${{ parameters.build.perf.cmake_args }} ${{ parameters.build.Virtual.cmake_args }} -DWORKER_THREADS=2"
+          cmake_env: "${{ parameters.build.Virtual.cmake_env }}"
+          suffix: "Perf_MultiThreaded"
+          artifact_name: "Virtual_Perf_MultiThreaded"
+          ctest_filter: "-R pi_basic_mt"
+          depends_on: configure
+
+  - ${{ if eq(parameters.perf_tests, 'run') }}:
+      - template: common.yml
+        parameters:
+          target: SGX
+          env: ${{ parameters.env.SGX }}
+          cmake_args: "${{ parameters.build.common.cmake_args }} ${{ parameters.build.perf.cmake_args }} ${{ parameters.build.SGX.cmake_args }} -DWORKER_THREADS=2"
+          suffix: "Perf_MultiThreaded"
+          artifact_name: "SGX_Perf_MultiThreaded"
+          ctest_filter: "-R pi_basic_mt"
+          depends_on: configure
```

Comment:
We're currently only building `-DWORKER_THREADS=2` in `.multi-thread.yml` (the `CCF Github Threading` pipeline). This adds some multi-threaded build+testing to the main CI pipeline. Are we confident that this is stable enough to promote?